app/models/mixins/virtual_total_mixin.rb

```diff
@@ -16,7 +16,7 @@ module ClassMethods
     #
     def virtual_total(name, relation, options = {})
       define_method(name) do
-        send(relation).try(:size) || 0
+        (attribute_present?(name) ? self[name] : nil) || send(relation).try(:size) || 0
```

Comment:
Sorry...I guess it *used* to be that way...seems ok now.

See attributes for the duping thing I was talking about (not sure if to_hash dups...don't know what object it is): https://github.com/rails/rails/blob/master/activerecord/lib/active_record/attribute_methods.rb#L275-L277